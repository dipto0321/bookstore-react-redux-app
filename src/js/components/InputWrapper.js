import React from 'react';

const InputWrapper = ({
  setInput, inputValue, labelValue, type, inputId, error,
}) => (
  <div className="input-wrapper">
    <label
      htmlFor={inputId}
      className="user-form__label"
    >
      { labelValue }
      {' '}
      {error}
    </label>
    <input
      type={type}
      id={inputId}
      className="form-input"
      value={inputValue}
      onChange={e => setInput(e.target.value)}
    />
  </div>
);

export default InputWrapper;
