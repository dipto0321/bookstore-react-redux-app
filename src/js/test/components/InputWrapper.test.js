import React from 'react';
import { shallow } from 'enzyme';
import InputWrapper from '../../components/InputWrapper';

describe('InputWrapper component', () => {
  test('should render correctly', () => {
    const setInput = jest.fn();
    const wrapper = shallow(
      <InputWrapper
        setInput={setInput}
        inputValue=""
        labelValue="Email/Username:"
        type="text"
        inputId="email"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
