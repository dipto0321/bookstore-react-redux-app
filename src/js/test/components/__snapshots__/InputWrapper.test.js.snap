// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWrapper component should render correctly 1`] = `
<div
  className="input-wrapper"
>
  <label
    className="user-form__label"
    htmlFor="email"
  >
    Email/Username:
  </label>
  <input
    className="form-input"
    id="email"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
